amp/WEB-INF/src/org/digijava/module/aim/util/ActivityVersionUtil.java

```diff
@@ -327,4 +337,396 @@ public static AmpActivityVersion initializeActivity(AmpActivityVersion act) thro
+ 	                            if (auxVersionable2.equalsForVersioning(auxVersionable1)) {
+ 	                                coincidence++;
+ 	                                Object auxValue1 = auxVersionable1.getValue() != null ? auxVersionable1.getValue() : "";
+ 	                                Object auxValue2 = auxVersionable2.getValue() != null ? auxVersionable2.getValue() : "";
+ 	                                if (!auxValue2.equals(auxValue1)) {
+ 	                                    // Check if the object was added in the
+ 	                                    // previous iteration.
+ 	                                    if (!auxList.contains(auxVersionable2)) {
+ 	                                        CompareOutput auxOutput = new CompareOutput(auxAnnotation.fieldTitle(),
+ 	                                                new String[]{
+ 	                                                        generateFormattedOutput(
+ 	                                                                auxVersionable1.getOutput()),
+ 	                                                        generateFormattedOutput(
+ 	                                                                auxVersionable2.getOutput())}, fields[i],
+ 	                                                new Object[]{auxObject1, auxObject2}, false, false);
+ 	                                        outputCollection.add(auxOutput);
+ 	                                    }
+ 	                                }
+ 	                            }
+ 	                        }
+ 	                        if (coincidence == 0) {
+ 	                            // Check if the object was added in the
+ 	                            // previous iteration.
+ 	                            if (!auxList.contains(auxVersionable2)) {
+ 	                                CompareOutput auxOutput = new CompareOutput(auxAnnotation.fieldTitle(), new String[]{
+ 	                                        "",
+ 	                                        generateFormattedOutput(auxVersionable2
+ 	                                                .getOutput())}, fields[i], new Object[]{null, auxObject2}, false,
+ 	                                        false);
+ 	                                outputCollection.add(auxOutput);
+ 	                            }
+ 	                        }
+ 	                    }
+ 	                } else {
+ 	                    output.setStringOutput(new String[]{auxResult1 != null ? auxResult1.toString() : "",
+ 	                            auxResult2 != null ? auxResult2.toString() : ""});
+ 	                    output.setDescriptionOutput(auxAnnotation.fieldTitle());
+ 	                    outputCollection.add(output);
+ 	                }
+ 	            }
+ 	        }
+
+ 	        return groupOutputCollection(outputCollection);
+ 	    }
+ 	    
+ 	   public static void compareToPrevious(Long activityOneId) throws Exception {
```

Comment:
This method will need to be called from a struts action. so it needs to return a value that will further be used by this struts action to render the comparison